### DISCUSSION TITLE
Better Install process

### DISCUSSION THREAD

**Discussion by JorgeCarousel** [apache/airflow#26213] [Ideas]
Hey guys,

I am researching for a good Project Management System, when I saw AirFlow, I said this could be it, but when I read how to install it all I could think was: "Why is it like that?" I have an Apache HTTPD Server running Python as CGI and MySQL as Database in place already I was wondering if a Web GUI for installation could be possible.

I personally don't like Java and all the others I have found are Java based, I thought AirFlow could be a game changer, but so far, I'm not impressed.

Can you make a Web GUI for installation?

Thank you.

**Comment by potiuk** [comment]
No. It's not possible, not our goal, you are likely not the target user if you are struggling with airflow installing via `pip` (this is the basic way how you install all Python packages).

**Comment by JorgeCarousel** [comment]
Thanks anyway, I just don't want to have multiple instances of Web servers running at once.

**Reply by potiuk** [reply]
> Thanks anyway, I just don't want to have multiple instances of Web servers running at once.

I am not sure where you have the notion that you have to, but maybe you misunderstood something or jump to conclusions without reading the docs first.

**Reply by potiuk** [reply]
But if you can point out to the docs that tell you that you have to have more than webserver instance - maybe you can point it out?

**Reply by potiuk** [reply]
I am all ears.

**Comment by potiuk** [comment]
And BTW. This is fine. You cannot satisfy everyone. But you know - Airlfow has 10s of thousands of users, and more than 2100 contributors - often people like you, who struggle with something. So if you think you can contribute a simpler way of installing Airlfow with Web GUI - you are absolutely welcome to try - as long as it does not break other experiences and does not require maintenance overhead - you can try to contribute it. I's all in your hands.

**Comment by JorgeCarousel** [comment]
I gave it another try on Windows with Python 3.10 and all I got was errors, for example, when I ran `airflow db init` I got this:

File "C:\Python310\lib\site-packages\airflow\www\app.py", line 84, in create_app
    raise AirflowConfigException(
airflow.exceptions.AirflowConfigException: Cannot use relative path: `sqlite:///C:\Users\Username/airflow/airflow.db` to connect to sqlite. Please use absolute path such as `sqlite:////tmp/airflow.db`.

And many more coming from other commands like `airflow standalone` or `airflow webserver --port 8080` so it doesn't start on Windows, it kind of worked on Linux on one of my tests, but what we are looking for is going to sit on a Windows Server.

**Reply by potiuk** [reply]
Airflow as you can read in the https://airflow.apache.org/docs/apache-airflow/stable/installation/prerequisites.html (which is good starting point to check for the installation capabilities) can only be run on Posix compliant OS's and for Windows, it runs on WSL2 only (or you can run it via Docker containers). 

> Airflow currently can be run on POSIX-compliant Operating Systems. For development it is regularly tested on fairly modern Linux Distros and recent versions of MacOS. On Windows you can run it via WSL2 (Windows Subsystem for Linux 2) or via Linux Containers. The work to add Windows support is tracked via [#10388](https://github.com/apache/airflow/issues/10388) but it is not a high priority. You should only use Linux-based distros as “Production” execution environment as this is the only environment that is supported. The only distro that is used in our CI tests and that is used in the [Community managed DockerHub image](https://hub.docker.com/p/apache/airflow) is Debian Bullseye.
 
We have an open issue and discussions on making Airlfow works on Windows natively -  https://github.com/apache/airflow/issues/10388 - initially for development (and we are close to it with finishing https://github.com/apache/airflow/projects/13 and merging this - git clone fix https://github.com/apache/airflow/pull/26105  and few other changes implemented by few other people in the past (for example https://github.com/apache/airflow/pull/24098). All of this is quite likely to happen in the coming month.

Production support for native Windows, however,  is far away. We are using many POSIX things (forking, processes, signal propagation, and a number of others that likely need thorough testing).

And if you feel like it - you are absolutely free to contribute any fixes and testing to move that forward. As mentioned before Airflow is created by more than 2100 contributors, it is a free software that the whole community contributes to (most of our contributors are users like you who have a problem and solve it). So feel free to join the efort and help to make Airflow more friendly for Windows users. This is our long term goal and you can help to make it reality.

**Reply by potiuk** [reply]
BTW. Making the problem with `airflow db init` on Windows might be a fantastic first contribution. That would be awesome if you could implement it.
